mirgecom/artificial_viscosity.py

```diff
@@ -103,37 +103,48 @@
 from grudge.dof_desc import DD_VOLUME_MODAL, DD_VOLUME
 
 
+def _get_actx(obj):
+    if isinstance(obj, TracePair):
+        return _get_actx(obj.int)
+    if isinstance(obj, np.ndarray):
+        return _get_actx(obj[0])
+    elif isinstance(obj, DOFArray):
+        return obj.array_context
+    else:
+        raise ValueError("Unknown type; can't retrieve array context.")
+
+
+# Tweak the behavior of np.outer to return a lower-dimensional object if either/both
+# of the arguments are scalars (np.outer always returns a matrix)
+def _outer(a, b):
+    if isinstance(a, np.ndarray) and isinstance(b, np.ndarray):
+        return np.outer(a, b)
+    else:
+        return a*b
```

Comment:
It'd be terrific to get these from:
`from mirgecom.operators import get_actx, outer`
or something like that.